autosklearn/ensemble_builder.py

```diff
@@ -21,33 +23,199 @@
+        event: str
+            The string identifier for an event indicating when this job must finish
+        backend: util.backend.Backend
+            backend to write and read files
+        dataset_name: str
+            name of dataset
+        task_type: int
+            type of ML task
+        metric: str
+            name of metric to score predictions
+        ensemble_size: int
+            maximal size of ensemble (passed to autosklearn.ensemble.ensemble_selection)
+        ensemble_nbest: int/float
+            if int: consider only the n best prediction
+            if float: consider only this fraction of the best models
+            Both wrt to validation predictions
+            If performance_range_threshold > 0, might return less models
+        max_models_on_disc: int
+           Defines the maximum number of models that are kept in the disc.
+           If int, it must be greater or equal than 1, and dictates the max number of
+           models to keep.
+           If float, it will be interpreted as the max megabytes allowed of disc space. That
+           is, if the number of ensemble candidates require more disc space than this float
+           value, the worst models will be deleted to keep within this budget.
+           Models and predictions of the worst-performing models will be deleted then.
+           If None, the feature is disabled.
+           It defines an upper bound on the models that can be used in the ensemble.
+        seed: int
+            random seed
+        max_iterations: int
+            maximal number of iterations to run this script
+            (default None --> deactivated)
+        precision: [16,32,64,128]
+            precision of floats to read the predictions
+        memory_limit: Optional[int]
+            memory limit in mb. If ``None``, no memory limit is enforced.
+        read_at_most: int
+            read at most n new prediction files in each iteration
+
+    Returns
+    -------
+        List[Tuple[int, float, float, float]]:
+            A list with the performance history of this ensemble, of the form
+            [[pandas_timestamp, train_performance, val_performance, test_performance], ...]
+    """
+
+    # Create the event as one of the criteria to decide if we move to he next iteration
+    event = dask.distributed.Event(event) if event else None
+
+    # The second criteria is elapsed time
+    elapsed_time = time.time() - start_time
+
+    # The last criteria is the number of iterations
+    iteration = 0
+
+    history = []
+
+    while True:
+
+        # First test for termination conditions
+        if time_left_for_ensembles < elapsed_time:
+            warnings.warn(
+                "Terminate ensemble building as not time is left (run for {}s)".format(
+                    elapsed_time
+                ),
+            )
+            break
+        if event is not None and event.is_set():
+            warnings.warn(
```

Comment:
We should not use warnings here as they might confuse a user. Can you somehow log to the regular logging file?

Reply:
I have added this to the ensemble log file and added a proper check for this!